### PR TITLE
Add TomTom GB spiders

### PR DESCRIPTION
**Pull request by Cj-Malone**
These have been written over the last few years, but I've just reviewed them and they are good to go!

### CODE REVIEW

**Comment by CloCkWeRX** [comment]
https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/benchmarx_gb.py#L13 - Q102181127 possible dupe?

**Comment by CloCkWeRX** [comment]
https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/nicholsons_pubs_gb.py#L8  Q113130666
Possible duplicate

**Comment by CloCkWeRX** [comment]
Would it be possible to maybe split this up a bit more, maybe in groups of 5-10? That'd get the ones with no issue CI results + merged quickly; while if there are duplicate spiders or similar ones that need a bit of discussion it only holds up a smaller number of new spiders?

File: locations/spiders/travelodge_gb.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import html
+
+from scrapy.spiders import SitemapSpider
+
+from locations.structured_data_spider import StructuredDataSpider
+
+
+class TravelodgeGBSpider(SitemapSpider, StructuredDataSpider):
+    name = "travelodge_gb"
+    item_attributes = {"brand": "Travelodge", "brand_wikidata": "Q9361374"}
```
Comment (CloCkWeRX) [inline locations/spiders/travelodge_gb.py:10]: Did we have this previously, then remove for some reason?

https://github.com/search?q=repo%3Aalltheplaces%2Falltheplaces+Q9361374&type=code

File: locations/spiders/mbplc_gb.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "Miller & Carter": {"brand": "Miller & Carter", "brand_wikidata": "Q87067401"},
+        "Nicholsons": {"brand": "Nicholsons", "brand_wikidata": "Q113130666"},
+        "O'Neill's": {"brand": "O'Neill's", "brand_wikidata": "Q7071905"},
+        "Premium Country Pubs": {
+            "brand": "Premium Country Pubs",
+            "brand_wikidata": "Q118606900",
+            "extras": Categories.PUB.value,
+        },
+        "Stonehouse": {"brand": "Stonehouse", "brand_wikidata": "Q78192049"},
+        "Toby Carvery": {"brand": "Toby Carvery", "brand_wikidata": "Q7811777"},
```
Comment (CloCkWeRX) [inline locations/spiders/mbplc_gb.py:25]: https://github.com/search?q=repo%3Aalltheplaces%2Falltheplaces+Q7811777&type=code - Q7811777 - 2x existing spiders to review

File: locations/spiders/sally_beauty_gb.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+from scrapy import Spider
+
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+
+
+class SallyBeautyGBSpider(Spider):
+    name = "sally_beauty_gb"
+    item_attributes = {"brand": "Sally Beauty", "brand_wikidata": "Q7405065"}
```
Comment (CloCkWeRX) [inline locations/spiders/sally_beauty_gb.py:11]: Possibly duplicate https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/sally_beauty.py#L14

File: locations/spiders/jet_gb.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+from locations.categories import Categories
+from locations.storefinders.storerocket import StoreRocketSpider
+
+
+class JetGBSpider(StoreRocketSpider):
+    name = "jet_gb"
+    item_attributes = {"brand": "JET", "brand_wikidata": "Q568940", "extras": Categories.FUEL_STATION.value}
```
Comment (CloCkWeRX) [inline locations/spiders/jet_gb.py:7]: Possible duplicate - https://github.com/search?q=repo%3Aalltheplaces%2Falltheplaces+Q568940&type=code - though this is probably the US version

File: locations/spiders/mbplc_gb.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    name = "mbplc_gb"
+    key = "woos-19f74f73-0b20-3571-b9ef-706ad3dbad27"
+    origin = "https://www.mbplc.com/"
+
+    brand_mapping = {
+        "Sizzling Pubs": {"brand": "Sizzling Pubs", "brand_wikidata": "Q118314187"},
+        "All Bar One": {"brand": "All Bar One", "brand_wikidata": "Q4728624"},
+        "Browns Restaurants": {"brand": "Browns Restaurants", "brand_wikidata": "Q4976672"},
+        "Ember Inns": {"brand": "Ember Inns", "brand_wikidata": "Q116272278"},
+        "Harvester": {"brand": "Harvester", "brand_wikidata": "Q5676915"},
```
Comment (CloCkWeRX) [inline locations/spiders/mbplc_gb.py:15]: https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/harvester_gb.py#L8

File: locations/spiders/essar_oil_gb.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+from locations.categories import Categories, Fuel, apply_category, apply_yes_no
+from locations.google_url import extract_google_position
+from locations.items import Feature
+
+
+# TODO: Address data not consistent between pages
+class EssarOilGBSpider(SitemapSpider):
+    name = "essar_oil_gb"
+    item_attributes = {"brand": "Essar", "brand_wikidata": "Q5399372"}
```
Comment (CloCkWeRX) [inline locations/spiders/essar_oil_gb.py:14]: https://github.com/search?q=repo%3Aalltheplaces%2Falltheplaces+Q5399372&type=code

But also probably just the US branch

File: locations/spiders/mbplc_gb.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+class MbplcGBSpider(WoosmapSpider):
+    name = "mbplc_gb"
+    key = "woos-19f74f73-0b20-3571-b9ef-706ad3dbad27"
+    origin = "https://www.mbplc.com/"
+
+    brand_mapping = {
+        "Sizzling Pubs": {"brand": "Sizzling Pubs", "brand_wikidata": "Q118314187"},
+        "All Bar One": {"brand": "All Bar One", "brand_wikidata": "Q4728624"},
+        "Browns Restaurants": {"brand": "Browns Restaurants", "brand_wikidata": "Q4976672"},
+        "Ember Inns": {"brand": "Ember Inns", "brand_wikidata": "Q116272278"},
```
Comment (CloCkWeRX) [inline locations/spiders/mbplc_gb.py:14]: https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/ember_inns_gb.py#L7 Q116272278

File: locations/spiders/inpost_gb.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+from scrapy.http import JsonRequest, Response
+
+from locations.categories import apply_yes_no
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+
+
+class InpostGBSpider(Spider):
+    name = "inpost_gb"
+    item_attributes = {"name": "InPost", "brand": "InPost", "brand_wikidata": "Q3182097"}
```
Comment (CloCkWeRX) [inline locations/spiders/inpost_gb.py:13]: TODO: Check the content of  https://github.com/alltheplaces/alltheplaces/issues/5912

**Comment by CloCkWeRX** [comment]
https://github.com/alltheplaces/alltheplaces/pull/9713

File: locations/spiders/mbplc_gb.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        "Harvester": {"brand": "Harvester", "brand_wikidata": "Q5676915"},
+        "Miller & Carter": {"brand": "Miller & Carter", "brand_wikidata": "Q87067401"},
```
Comment (CloCkWeRX) [inline locations/spiders/mbplc_gb.py:16]: https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/miller_and_carter_gb.py#L7 Q87067401

File: locations/spiders/mbplc_gb.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+        "Nicholsons": {"brand": "Nicholsons", "brand_wikidata": "Q113130666"},
+        "O'Neill's": {"brand": "O'Neill's", "brand_wikidata": "Q7071905"},
+        "Premium Country Pubs": {
+            "brand": "Premium Country Pubs",
+            "brand_wikidata": "Q118606900",
+            "extras": Categories.PUB.value,
+        },
+        "Stonehouse": {"brand": "Stonehouse", "brand_wikidata": "Q78192049"},
```
Comment (CloCkWeRX) [inline locations/spiders/mbplc_gb.py:24]: https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/stonehouse_gb.py#L8 Q78192049

File: locations/spiders/mbplc_gb.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    origin = "https://www.mbplc.com/"
+
+    brand_mapping = {
+        "Sizzling Pubs": {"brand": "Sizzling Pubs", "brand_wikidata": "Q118314187"},
+        "All Bar One": {"brand": "All Bar One", "brand_wikidata": "Q4728624"},
+        "Browns Restaurants": {"brand": "Browns Restaurants", "brand_wikidata": "Q4976672"},
+        "Ember Inns": {"brand": "Ember Inns", "brand_wikidata": "Q116272278"},
+        "Harvester": {"brand": "Harvester", "brand_wikidata": "Q5676915"},
+        "Miller & Carter": {"brand": "Miller & Carter", "brand_wikidata": "Q87067401"},
+        "Nicholsons": {"brand": "Nicholsons", "brand_wikidata": "Q113130666"},
```
Comment (CloCkWeRX) [inline locations/spiders/mbplc_gb.py:17]: https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/nicholsons_pubs_gb.py#L8  Q113130666

File: locations/spiders/travis_perkins_gb.py [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+from locations.dict_parser import DictParser
+from locations.hours import OpeningHours
+from locations.pipelines.address_clean_up import merge_address_lines
+
+
+class TravisPerkinsGBSpider(Spider):
+    name = "travis_perkins_gb"
+    item_attributes = {"brand": "Travis Perkins", "brand_wikidata": "Q2450664"}
+    BENCHMARX = {"brand": "Benchmarx", "brand_wikidata": "Q102181127"}
```
Comment (CloCkWeRX) [inline locations/spiders/travis_perkins_gb.py:14]: https://github.com/alltheplaces/alltheplaces/blob/f7eb8baf14984461fadb5993e545019706d8b135/locations/spiders/benchmarx_gb.py#L13 - Q102181127

File: locations/spiders/fitness_first_gb.py [hunk 13]
```diff
@@ -0,0 +1,18 @@
+from scrapy.spiders import SitemapSpider
+
+from locations.google_url import extract_google_position
+from locations.structured_data_spider import StructuredDataSpider
+
+
+class FitnessFirstGBSpider(SitemapSpider, StructuredDataSpider):
+    name = "fitness_first_gb"
+    item_attributes = {"brand": "Fitness First", "brand_wikidata": "Q127120"}
```
Comment (CloCkWeRX) [inline locations/spiders/fitness_first_gb.py:9]: https://github.com/search?q=repo%3Aalltheplaces%2Falltheplaces+Q127120&type=code - Q127120
Possibly international versions/non issue

Comment (CloCkWeRX) [reply locations/spiders/fitness_first_gb.py:9]: Okay, non issue

File: locations/spiders/essar_oil_gb.py (hunk 6 above)
Comment (CloCkWeRX) [reply locations/spiders/essar_oil_gb.py:14]: Non issue

File: locations/spiders/sally_beauty_gb.py (hunk 3 above)
Comment (CloCkWeRX) [reply locations/spiders/sally_beauty_gb.py:11]: Okay, existing spider produces 0 results
https://alltheplaces-data.openaddresses.io/runs/2024-08-17-13-32-02/logs/sally_beauty.txt

File: locations/spiders/travis_perkins_gb.py (hunk 12 above)
Comment (CloCkWeRX) [reply locations/spiders/travis_perkins_gb.py:14]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fbenchmarx_gb.geojson - 30 results, but has dropped from 128 in the last month

File: locations/spiders/inpost_gb.py (hunk 8 above)
Comment (CloCkWeRX) [reply locations/spiders/inpost_gb.py:13]: Okay, non issue - existing is PL only
https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fpaczkomat_inpost_pl.geojson

File: locations/spiders/jet_gb.py (hunk 4 above)
Comment (CloCkWeRX) [reply locations/spiders/jet_gb.py:7]: Non issue

File: locations/spiders/mbplc_gb.py (hunk 7 above)
Comment (CloCkWeRX) [reply locations/spiders/mbplc_gb.py:14]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fember_inns_gb.geojson - 148 covered

File: locations/spiders/mbplc_gb.py (hunk 5 above)
Comment (CloCkWeRX) [reply locations/spiders/mbplc_gb.py:15]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fharvester_gb.geojson - current spider has 0 output

File: locations/spiders/mbplc_gb.py (hunk 9 above)
Comment (CloCkWeRX) [reply locations/spiders/mbplc_gb.py:16]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fmiller_and_carter_gb.geojson - 129 locations

File: locations/spiders/mbplc_gb.py (hunk 11 above)
Comment (CloCkWeRX) [reply locations/spiders/mbplc_gb.py:17]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fnicholsons_pubs_gb.geojson 78 locations

File: locations/spiders/mbplc_gb.py (hunk 10 above)
Comment (CloCkWeRX) [reply locations/spiders/mbplc_gb.py:24]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Fstonehouse_gb.geojson 92 locations

File: locations/spiders/mbplc_gb.py (hunk 2 above)
Comment (CloCkWeRX) [reply locations/spiders/mbplc_gb.py:25]: https://alltheplaces-data.openaddresses.io/map.html?show=https%3A%2F%2Falltheplaces-data.openaddresses.io%2Fruns%2F2024-08-17-13-32-02%2Foutput%2Ftoby_carvery_gb.geojson - 150 locations

File: locations/spiders/sally_beauty_gb.py (hunk 3 above)
Comment (Cj-Malone) [reply locations/spiders/sally_beauty_gb.py:11]: US/CA vs GB

**Comment by Cj-Malone** [comment]
I don't consider these duplicates. Yea some cover the same brands, but they are different sources, they have different attributes, different levels of quality and freshness.

**Comment by CloCkWeRX** [comment]
@iandees what helps get this merged? Can do more chunks so that CI runs or split off PRs to replace currently busted spiders.

@Cj-Malone tell me to shoo or close the other PRs if it's accidentally more of a hindrance than a help!

**Comment by iandees** [comment]
The smaller chunks of spiders are easier to review for sure, thanks for doing those @CloCkWeRX. I saw that there's a part 1 and part 3, so I was waiting for a part 2 before looking too close. If you're all set then I will look.

**Comment by CloCkWeRX** [comment]
Ah part 2 was https://github.com/alltheplaces/alltheplaces/pull/9739 but didn't quite make it.

**Comment by iandees** [comment]
Ok I think I'm going to merge this because I saw the runs in @CloCkWeRX's split up PRs.

**Comment by Cj-Malone** [comment]
Thank you

**Pull request merged**